tests/e2e/dpa_deployment_suite_test.go

```diff
@@ -661,5 +480,132 @@ var _ = Describe("Configuration testing for DPA Custom Resource", func() {
 			},
 			WantError: true,
 		}, fmt.Errorf("region for AWS backupstoragelocation cannot be empty when s3ForcePathStyle is true or when backing up images")),
+	}
+
+	if provider == "aws" {
+		genericTests = append(genericTests, awsTests...)
+	}
+	DescribeTable("Updating custom resource with new configuration",
+
+		func(installCase InstallCase, expectedErr error) {
+			//TODO: Calling dpaCR.build() is the old pattern.
+			//Change it later to make sure all the spec values are passed for every test case,
+			// instead of assigning the values in advance to the DPA CR
+			err := dpaCR.Build(installCase.BRestoreType)
+			Expect(err).NotTo(HaveOccurred())
+			if len(installCase.DpaSpec.BackupLocations) > 0 {
+				switch dpaCR.Provider {
+				case "aws":
```

Comment:
You can feed this data to the json instead, and then you won't have to switch over the provider type; this way it will be initialized generically inside dpaCr.build function:
```
					Velero: &velero.BackupStorageLocationSpec{
						Provider:   v.CustomResource.Spec.BackupLocations[0].Velero.Provider,
						Default:    true,
						Config:     v.CustomResource.Spec.BackupLocations[0].Velero.Config,
						Credential: v.CustomResource.Spec.BackupLocations[0].Velero.Credential,
```